cloudinit/distros/__init__.py

```diff
@@ -749,6 +752,22 @@ def create_group(self, name, members=None):
                 subp.subp(['usermod', '-a', '-G', name, member])
                 LOG.info("Added user '%s' to group '%s'", member, name)
 
+    def shutdown_command(self, mode='poweroff', delay='now', message=None):
+        # called from cc_power_state_change.load_power_state
+        command = ["shutdown", self.shutdown_options_map[mode]]
+        try:
+            if delay != "now":
+                delay = "+%d" % int(delay)
+        except ValueError as e:
+            raise TypeError(
+                "power_state[delay] must be 'now' or '+m' (minutes)."
+                " found '%s'." % delay
```

Comment:
We can't be sure of `delay`s type in this exception codepath (explicitly, because we're handling a `ValueError`), so we should [defend against `delay` being a tuple](https://write.wrestle.town/oddbloke/python-anti-pattern-single-element-s-formatting):

```suggestion
                " found '%s'." % (delay,)
```